gettsim/pre_processing/policy_for_date.py

```diff
@@ -48,62 +50,172 @@
-            raise ValueError(f"{groups} is not a category for groups.")
-    else:
-        raise ValueError(f"{groups} is not a string or list.")
+def get_policies_for_date(policy_date, policy_groups="all"):
+    """Get the state of the policy system for a particular date.
+
+    The function returns time dependent policy reforms (functions) as well as the
+    policy parameters for the current policy system.
+
+    Parameters
+    ----------
+    policy_date : int, str, datetime.date
+        The date for which the policy system is set up.
+    policy_groups : list, str
+        The group or a list of groups which parameters are loaded. If an invalid
+        name is given, a list of all possible values is printed. Default is to load all
+        parameter groups.
+
+    Returns
+    -------
+    params_dict : dict
+        Dictionary of parameters grouped in policy system compartments given in groups.
+    policy_func_dict : dict
+        Dictionary of time dependent policy reforms. Keys are the variable names they
+        create.
+
+    """
+    # Check policy date for correct format and transfer to datetime.date
+    policy_date = _parse_date(policy_date)
+
+    # Check groups argument for correct format and transfer to list.
+    group_list = _parse_parameter_groups(policy_groups)
 
     params_dict = {}
-    year = policy_date.year
 
     for group in group_list:
-        tax_data = load_data(policy_date, group)
-
-        for param in tax_data:
-            if type(tax_data[param]) == dict:
-                if "type" in tax_data[param]:
-                    if tax_data[param]["type"].startswith("piecewise"):
-                        if "progressionsfaktor" in tax_data[param]:
-                            if tax_data[param]["progressionsfaktor"]:
-                                tax_data[param] = add_progressionsfaktor(
-                                    tax_data[param], param
-                                )
-                        tax_data[param] = get_piecewise_parameters(
-                            tax_data[param],
-                            param,
-                            func_type=tax_data[param]["type"].split("_")[1],
-                        )
-                for key in ["type", "progressionsfaktor"]:
-                    tax_data[param].pop(key, None)
+        tax_data = _load_parameter_group_from_yaml(policy_date, group)
+
+        # Align paramters for e.g. piecewise polynomial functions
+        params_dict[group] = align_parameters(tax_data)
+
+    policy_func_dict = load_reforms_for_date(policy_date)
 
-        tax_data["datum"] = policy_date
-        params_dict[group] = tax_data
+    return params_dict, policy_func_dict
+
+
+def align_parameters(tax_data):
+    """Check if parameters are stored in implicit structures and align to general
+    structure.
+
+    Parameters
+    ----------
+    tax_data : dict
+        Loaded raw tax data.
+
+    Returns
+    -------
+
+    """
+    for param in tax_data:
```

Comment:
`tax_data.items()`?

Reply:
No, I need to operate on the original dict and therefore always select by name.